src/common/src/types/interval.rs

```diff
@@ -81,23 +81,40 @@ impl IntervalUnit {
         self.usecs.rem_euclid(USECS_PER_DAY) as u64
     }
 
-    /// Justify interval, convert 1 month to 30 days and 86400 s to 1 day.
-    /// If day is positive, complement the seconds negative value.
-    /// These rules only use in interval comparison.
-    pub fn justify_interval(&mut self) {
-        #[expect(deprecated)]
-        let total_usecs = self.as_usecs_i64();
-        *self = Self {
-            months: 0,
-            days: (total_usecs / USECS_PER_DAY) as i32,
-            usecs: total_usecs % USECS_PER_DAY,
+    /// Justify interval, convert 24 hours to 1 day and 30 days to 1 month.
+    /// Also makes signs of all fields to be the same.
+    ///
+    /// <https://github.com/postgres/postgres/blob/REL_15_2/src/backend/utils/adt/timestamp.c#L2740>
+    pub fn justify_interval(&self) -> Option<Self> {
+        let mut v = *self;
+        if v.days > 0 && v.usecs > 0 || v.days < 0 && v.usecs < 0 {
```

Comment:
I don't really understand this, why do we check `usecs` here also. What does the conditional expression here check for?

Reply:
Added comments for this.

Actually identified a "bug" for a line below, which may report overflow unnecessarily. The bug is even [present](https://www.db-fiddle.com/f/jvRiNUTFALCUTCtgLNL2fm/0) in [latest PostgreSQL](https://github.com/postgres/postgres/blame/master/src/backend/utils/adt/timestamp.c#L2768).

To clarify, the new comparison logic is no longer depending on this `justify_interval`. It is retained solely for future exposure as a SQL function, which is not the main goal of this PR.

Now talking about this `justify_interval`, I think we can:
* Keep this bug-compatible version.
* Fix the bug on top of current implementation.
* Remove this duplicated implementation and just use `IntervalCmpValue::as_justified`, which is slower.

I am thinking to deferred the decision until we expose `justify_interval`. I can remove this function completely in this PR if that is preferred.

Reply:
> To clarify, the new comparison logic is no longer depending on this justify_interval. It is retained solely for future exposure as a SQL function, which is not the main goal of this PR.

Sounds good, thanks for clarifying.

> I am thinking to deferred the decision until we expose justify_interval. I can remove this function completely in this PR if that is preferred.

Is `justify_interval` still used anywhere in our code by `Interval`? If it isn't that is fine, we can defer.
But if it is, I'm wondering if the difference in `justify_interval` and `as_justified` can lead to different representations of interval. This can lead to wrong hash values if they follow different code paths.

Reply:
> still used anywhere

No. And I think removing it completely in this PR could help making it clear.

> if the difference in `justify_interval` and `as_justified` can lead to different representations of interval. This can lead to wrong hash values if they follow different code paths.

I will remove the former and add it back later as a pure user-facing function.

Furthermore, the correctness of Eq/Hash/Ord does not rely on them giving the same result - they could just return completely different results, for example deserialize from memcomparable or HashKey could always return the alternate form. The `as_justified` is just more "intuitive" thus preferred when available. But we can drop it if it is actually causing more confusion. For now I am going to update its description rather than remove it together.

Reply:
Okay that works 👍